chromium_src/third_party/blink/renderer/modules/canvas/canvas2d/base_rendering_context_2d.cc

```diff
@@ -8,13 +8,32 @@
 #include "base/notreached.h"
 #include "brave/third_party/blink/renderer/core/farbling/brave_session_cache.h"
 #include "third_party/blink/renderer/platform/graphics/image_data_buffer.h"
+#include "third_party/blink/renderer/platform/weborigin/kurl.h"
 
-#define BRAVE_GET_IMAGE_DATA                                                  \
-  if (ExecutionContext* context = ExecutionContext::From(script_state)) {     \
-    SkPixmap image_data_pixmap = image_data->GetSkPixmap();                   \
-    brave::BraveSessionCache::From(*context).PerturbPixels(                   \
-        static_cast<const unsigned char*>(image_data_pixmap.writable_addr()), \
-        image_data_pixmap.computeByteSize());                                 \
+namespace blink {
+namespace {
+
+bool IsGoogleMaps(const KURL& url) {
+  const auto host = url.Host();
+  if (!host.StartsWith("google.") && !host.Contains(".google.")) {
+    return false;
+  }
+  const auto path = url.GetPath();
+  return path.StartsWith("/maps/") || path == "/maps";
+}
+
+}  // namespace
+}  // namespace blink
+
+#define BRAVE_GET_IMAGE_DATA                                              \
+  if (ExecutionContext* context = ExecutionContext::From(script_state)) { \
```

Comment:
nit `if (!IsGoogleMaps(context->Url()) && ExecutionContext* context = ExecutionContext::From(script_state))` ?

Reply:
I need to declare `context` first. :)

I tried combining the two things in the opposite order in a single `if` statement, but the compiler didn't like it.